UnityPython.BackEnd/src/Traffy.Objects/List.cs

```diff
@@ -96,4 +103,191 @@ void TrObject.__setitem__(TrObject item, TrObject value)
         }
     }
 
+    public partial class TrMapObject : TrObject, IEnumerator<TrObject>
+    {
+        public static TrClass CLASS;
+        public TrClass Class => CLASS;
+
+        TrObject func;
+        IEnumerator<TrObject> gen;
+
+        public TrMapObject(TrObject func, List<IEnumerator<TrObject>> items)
+        {
+            this.func = func;
+            this.gen = generator(func, items);
+        }
+
+        [Mark(Initialization.TokenClassInit)]
+        static void _Init()
+        {
+            CLASS = TrClass.FromPrototype<TrMapObject>();
+            CLASS.Name = "map";
+            CLASS.InitInlineCacheForMagicMethods();
+            CLASS[CLASS.ic__new] = TrStaticMethod.Bind("map.__new__", TrMapObject.datanew);
+            CLASS.IsSealed = true;
+            TrClass.TypeDict[typeof(TrMapObject)] = CLASS;
+        }
+
+        [Mark(typeof(TrMapObject))]
+        static void _SetupClasses()
+        {
+            CLASS.SetupClass();
+            CLASS.IsFixed = true;
+            Initialization.Prelude(CLASS);
+        }
+
+        public static TrObject datanew(TrClass cls, BList<TrObject> args, Dictionary<TrObject, TrObject> kwargs)
+        {
+            int len = args.Count;
+            if (len < 2)
+            {
+                throw new TypeError("map() must have at least two arguments.");
+            }
+
+            var list = new List<IEnumerator<TrObject>>();
```

Comment:
size can be statically decided, use array instead, if you have time for this minor thing.